packages/NodeNameResolver/Contract/NodeNameResolverInterface.php

```diff
@@ -19,5 +20,5 @@ public function getNode(): string;
     /**
      * @param TNode $node
      */
-    public function resolve(Node $node): ?string;
+    public function resolve(Node $node, ?Scope $scope): ?string;
```

Comment:
first step is to allow a optional scope here. after we cleaned up all call-sites we can finally make this non-nullable (several steps across multiple PRs)